### ISSUE TITLE
CORS error from token endpoint

### ISSUE THREAD

**Issue by youjeongsue** [keycloak/keycloak#15744]
### Area

authentication

### Describe the bug

I am trying to log in using ReactKeycloakProvider in react.
When I try to log in from localhost:3000, it fails with the following error.
I googled a lot but couldn't solve this. TT

I set the web origin correctly in my client settings.
but I don't know why I am getting this error on the token endpoint.

`
from origin 'http://localhost:3000' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.
`

<img width="655" alt="image" src="https://user-images.githubusercontent.com/43227174/204526531-a9b59b21-94f4-46ed-9b97-01ddf70795e7.png">

### Version

12.0.4

### Expected behavior

I want to log in successfully.

### Actual behavior

cors error above

### How to Reproduce?

in my react code
```
keycloak-js: ^20.0.0
@react-keycloak/web: ^3.2.1
```
```js
import Keycloak from "keycloak-js";

const keycloak = new Keycloak({
 url: "https://my-domain.com/auth",
 realm: "my-realm",
 clientId: "React-auth"
});

const initOptions = {
  onLoad: 'login-required'
};
```

And the keycloak client is set as follows.
<img width="1108" alt="image" src="https://user-images.githubusercontent.com/43227174/204527388-7bb28e1c-60f1-4346-877c-8105d97dbb62.png">

After trying to log in, the following log is also left in the keycloak event. But there is no error log.
<img width="1191" alt="image" src="https://user-images.githubusercontent.com/43227174/204527785-98c898ad-d0d6-4e75-9190-5f2573b8f321.png">

### Anything else?

I've already seen the same issue, but none of the 3 cases mentioned here seemed to apply to my case.
https://github.com/keycloak/keycloak/issues/14980

ex) my realm ssl setting
<img width="496" alt="image" src="https://user-images.githubusercontent.com/43227174/204529285-380aa11b-ccfc-4b47-867e-db8d0a1995c1.png">

How can I solve this? Could I be missing something? Any help would be greatly appreciated!!

**Comment by jbman** [comment]
Why does ReactKeycloakProvider set `XMLHttpRequest.withCredentials` to `true`?
Default is `false`, see https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/withCredentials
If it is set to `true`, cookies could be send to the token endpoint which doesn't expect or sets any cookies.

Note: Not a Keycloak bug.

**Comment by youjeongsue** [comment]
Thank you so much for your reply!!

But I didn't do any additional setting for `withCredential`.
How can I reset ReactKeycloakProvider's `withCredential` to false?

**Comment by mposolda** [comment]
@jbman Thanks for your statement on this! I am closing as the GH issues are dedicated for Keycloak bugs.
@youjeongsue Please use user forums for asking questions.

**Comment by ignitehub-dev** [comment]
@mposolda you should have included a link to the new issue for people with the same issue ...
